packages/create-cloudflare/src/frameworks/next/index.ts

```diff
@@ -72,16 +73,74 @@ const configure = async (ctx: PagesGeneratorContext) => {
 	writeFile(handlerPath, handlerFile);
 	updateStatus("Created an example API route handler");
 
+	const installEslintPlugin = await shouldInstallNextOnPagesEslintPlugin(ctx);
+
+	if (installEslintPlugin) {
+		await writeEslintrc(ctx);
+		updateStatus("eslint-plugin-next-on-pages added to eslintrc");
+	}
+
 	// Add some dev dependencies
 	process.chdir(projectName);
-	const packages = ["@cloudflare/next-on-pages@1", "vercel"];
+	const packages = [
+		"@cloudflare/next-on-pages@1",
+		"vercel",
+		...(installEslintPlugin ? ["eslint-plugin-next-on-pages"] : []),
+	];
 	await installPackages(packages, {
 		dev: true,
 		startText: "Adding the Cloudflare Pages adapter",
 		doneText: `${brandColor(`installed`)} ${dim(packages.join(", "))}`,
 	});
 };
 
+// Note: this isn't a generic helper since it's Next specific, this can be
+//       made into a generic helper by checking all possible types of configs
+//       (including a field in the package.json)
+//       (and potentially returning what type of config is used)
```

Comment:
Is this ok? or should we try to make it generic out of the bat?

Reply:
mh.... I followed what usesTypescript does:
https://github.com/cloudflare/workers-sdk/blob/b58663a8762248465cc7c35df7d7030a0858aa8d/packages/create-cloudflare/src/helpers/files.ts#L45-L47

is that not correct?

Reply:
I think that also might be a potential bug. I think the helper functions shouldn't rely on the guarantee that `.` matches `cwd` at any one point in time, because there are a few cases where it doesn't. If you think about the lifecycle of a c3 run, it creates a directory via some framework tool, does some configuration, and typically only then does it `cd` into the created project directory. This means that if you write some code that invokes one of these helpers in the configuration step, you will get a different result than you would in the deploy step.

Reply:
shall I update also the `usesTypescript` code? or do you prefer that to be done later/separately?